collect_app/src/main/java/org/odk/collect/android/utilities/SoftKeyboardController.kt

```diff
@@ -1,15 +1,21 @@
 package org.odk.collect.android.utilities
 
 import android.content.Context
+import android.os.Handler
+import android.os.Looper
 import android.view.View
 import android.view.inputmethod.InputMethodManager
 
 object SoftKeyboardController {
     fun showSoftKeyboard(view: View) {
-        if (view.requestFocus()) {
-            val inputMethodManager = view.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-            inputMethodManager.showSoftInput(view, 0)
-        }
+        // Subtle delay in displaying the keyboard is necessary for the keyboard to be displayed at all,
+        // for example, when returning from the hierarchy view.
+        Handler(Looper.getMainLooper()).postDelayed({
```

Comment:
Were you able to experiment shorter delays than 100ms here or even just adding no time based delay (with just a `post`)?

Also, we should probably use `Scheduler` here rather than referencing `Handler` directly here.

Reply:
> Were you able to experiment shorter delays than 100ms here or even just adding no time based delay (with just a post)?

I think I did and maybe it even worked but wanted to use a delay similar to what we have in other places (see above) to not risk that on some devices it wouldn't be enough.

> Also, we should probably use Scheduler here rather than referencing Handler directly here.

As I said it's rather a temporary solution so I didn't want to implement too many changes but passing `scheduler` to SoftKeyboardController would require more work.

Reply:
Ah you mean that we'd get rid of this after #5853? I'll make sure that's in the next milestone.